packages/shared/src/db/queries.ts

```diff
@@ -153,6 +153,32 @@ export const getGroupPreviews = createReadQuery(
   ['groups']
 );
 
+export const getJoinedGroupsCount = createReadQuery(
+  'getJoinedGroupCount',
+  async (ctx: QueryCtx) => {
+    const joinedGroups = await ctx.db.query.groups.findMany({
+      where: eq($groups.currentUserIsMember, true),
+    });
+    return joinedGroups.length;
+  },
+  ['groups']
+);
+
+export const getGroupsWithMemberThreshold = createReadQuery(
+  'getGroupsWithMemberThreshold',
+  async (threshold: number, ctx: QueryCtx) => {
```

Comment:
This could also be done a lot more efficiently -- check out `unreadCounts` filtering in `getGroups`.